src/extension.ts

```diff
@@ -14,8 +14,16 @@ export function activate(context: ExtensionContext) {
         sonicPi.stopAllCode();
     });
 
+    let restartCmd = commands.registerCommand('extension.restart', () => {
+        sonicPi.stopAllCode();
+        setTimeout(() => {
+            sonicPi.runCode();
+        }, 500);
```

Comment:
It will work, but I don't like this way because it cause unexpected result when `sonicPi.stopAllCode()` take a over 500ms for some reasons. 
I'll look for a way to know `sonicPi.stopAllCode()` is done.  If it is difficult to know that, I want adopt this way for now.

Reply:
Is there a way to delay only if sonic is running?

Reply:
I can't have a time to look for the way, so I'll merge this code for now.